### PR TITLE
Mutating Azure server group operations now handle either `regions` or…

### PR DESCRIPTION
**Pull request by duftler**
… `region`.
`regions` is deprecated in favor of `region`.
@milismsft please review.

### CODE REVIEW

**Comment by duftler** [comment]
Make sure to view the diff for this pr with `?w=1` like so: https://github.com/spinnaker/clouddriver/pull/542/files?w=1

Otherwise, the unimportant whitespace delta (from removing the surrounding `regions` loops in the 3 operations) will drown out the important details.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/EnableDisableDestroyAzureServerGroupDescription.groovy [hunk 1]
```diff
@@ -18,5 +18,8 @@ package com.netflix.spinnaker.clouddriver.azure.resources.servergroup.model
 
 class EnableDisableDestroyAzureServerGroupDescription extends AzureServerGroupDescription{
   String serverGroupName
+  String region
```
Comment (milismsft) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/EnableDisableDestroyAzureServerGroupDescription.groovy]: region is already defined in the base class AzureResourceOpsDescription (all our descriptions are expected to extend that class)

Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/EnableDisableDestroyAzureServerGroupDescription.groovy]: Ah, missed that. I've fixed it and re-pushed the branch.

**Comment by milismsft** [comment]
How does RegionsToRegionDescriptionPreProcessor come in play? Don't we need to call it in the converter classes (i.e. DestroyAzureServerGroupAtomicOperationConverter)?

Looking at the code we also use regions for some of the operations on the security group and load balancer. Do we need to change those as well?

**Comment by duftler** [comment]
You can see in this PR (https://github.com/spinnaker/clouddriver/pull/235) how this code is called in `OperationsController` just before the validators are called.

> Looking at the code we also use regions for some of the operations on the security group and load balancer. Do we need to change those as well?

Nope. These changes specifically relate to coming changes in orca that are contained to the mutating server group operations.

**Comment by milismsft** [comment]
LGTM then :-)

**Pull request merged**